src/build/plan.rs

```diff
@@ -206,14 +206,11 @@ impl Plan {
         while let Some(top) = to_process.pop() {
             if transitive.get(&top).is_some() {
                 continue;
-            } else {
-                transitive.insert(top.clone());
             }
+            transitive.insert(top.clone());
 
             // Process every dirty rev dep of the processed node
-            let dirty_rev_deps = self.rev_dep_graph
-                .get(&top)
-                .unwrap()
+            let dirty_rev_deps = self.rev_dep_graph[&top]
```

Comment:
Coming from C++, this makes me a little more than uneasy. There, it did tend to introduce very subtle bugs due to its `entry(key).or_default()` semantics in `map`s, rather than `get(key).unwrap()`. I realize that it's a different language with different semantics, thankfully, so it's probably fine here - my inner C++ self will have to get used to it 😄

Reply:
It seems nicer to me, but I mostly stayed away from C++!